### DISCUSSION TITLE
Up frame element suggestion

### DISCUSSION THREAD

**Discussion by acetinick** [unpoly/unpoly#231] [Ideas]
One of the things that i like about the turbo Hotwire framework is their use of a turbo-frame element that treats all links within frame as a turbo request.

Unpoly could really benefit from this approach by simply wrapping sections of html with something like an up-frame element that will treat all links within frame as up links. Huge benefits for things like data grids with sorting and pagination etc.

thoughts?

**Comment by adam12** [comment]
Interesting idea.

I think you could roll your own fairly easily, with `up.macro`. I whipped [this](https://glitch.com/edit/#!/shore-fourth-fisherman) up in a few minutes so it might be buggy but it demonstrates what I was thinking.

**Reply by acetinick** [reply]
Nice! never seen up.macro before, ill take a look. 

There are some cases need to think about such as, there will be cases you want to force the links not to follow,  for example clicking link to record in the datatable you would expect to do a full page load.  

Turbo handles this via using data-turbo-frame="_top" to treat it as a normal link.  Not the best syntax, but yea.
